### PR TITLE
[MRG] Refactor reporting.py module as directory to prepare for incorporating GLM Reporter

### PR DESCRIPTION
**Pull request by kchawla-pi**
@bthirion As you suggested and A Abadie mentioned, making piecemeal changes over multiple PRs. This is straight forward. Once this is merged, We can work on the GLM Reporter PR again.

### CODE REVIEW

File: nistats/reporting/_compare_niimgs.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                   ref_label="image set 1", src_label="image set 2",
+                   output_dir=None, axes=None):
+    """Creates plots to compare two lists of images and measure correlation.
+
+    The first plot displays linear correlation between voxel values
+    The second plot superimposes histograms to compare values distribution
+
+    Parameters
+    ----------
+    ref_imgs: nifti_like
```
Comment (bthirion) [inline nistats/reporting/_compare_niimgs.py]: The docstring is not consistent with the function signature...

File: nistats/reporting/_get_clusters_table.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+def _local_max(data, affine, min_distance):
+    """Find all local maxima of the array, separated by at least min_distance.
+    Adapted from https://stackoverflow.com/a/22631583/2589328
+
+    Parameters
+    ----------
+    data : array_like
+        3D array of with masked values for cluster.
+
```
Comment (bthirion) [inline nistats/reporting/_get_clusters_table.py]: incomplete docstring

File: nistats/reporting/_compare_niimgs.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    output_dir: string, optional (default None)
+        Directory where plotted figures will be stored.
+
+    axes: list of two matplotlib Axes objects, optional (default None)
+        Can receive a list of the form [ax1, ax2] to render the plots.
+        By default new axes will be created
+
+    Returns
+    -------
+    Pearson correlation between the images
```
Comment (bthirion) [inline nistats/reporting/_compare_niimgs.py]: corrs: ndarray,

File: nistats/reporting/_get_clusters_table.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    vals = vals[order]
+    ijk = ijk[order, :]
+    xyz = nib.affines.apply_affine(affine, ijk)  # Convert to xyz in mm
+    
+    # Reduce list of subpeaks based on distance
+    keep_idx = np.ones(xyz.shape[0]).astype(bool)
+    for i in range(xyz.shape[0]):
+        for j in range(i + 1, xyz.shape[0]):
+            if keep_idx[i] == 1:
+                dist = np.linalg.norm(xyz[i, :] - xyz[j, :])
```
Comment (bthirion) [inline nistats/reporting/_get_clusters_table.py]: This loops seems pretty inefficient if xyz.shape[0] is large. there is room for improvement here.

Comment (kchawla-pi) [reply nistats/reporting/_get_clusters_table.py]: Based on limited profiling the speed hit is minimal.
Not rally sure how to optimize this in a meaningful way. 
Looking into it.

**Comment by kchawla-pi** [comment]
I have incorporated the feedback (except one). Anything more?

**Pull request merged**